### PR TITLE
feat: add single video & history page

### PR DESCRIPTION
**Pull request by PuneetP16**
Features Added:
1. responsive single video page and history page.
2. add video to history alert

image
image

### CODE REVIEW

File: src/components/Cards/VideoCard/VideoCard.jsx [hunk 1]
```diff
@@ -45,18 +46,20 @@ export const VideoCard = ({ video }) => {
 		if (!isAuth) navigate("/login");
 	};
 
-	// console.log({
```
Comment (Ananyamadhu08) [inline src/components/Cards/VideoCard/VideoCard.jsx:48]: you can delete unnecessary commented code.

**Review by Ananyamadhu08 — APPROVED** [review]
Hey Puneet! Good job with the UI, your code is structured and neat. You can delete the commented code, other than that your code looks good to go you can go ahead and merge!

File: src/components/UI/Alert/Alert.jsx [hunk 2]
```diff
@@ -23,16 +26,16 @@ export const Alert = ({ alert, setAlert }) => {
...
-			<div className={`alert ${type} alert--dismissable`}>
-				{text}
-				<button
-					onClick={onClickHandler}
-					className="btn btn--icon btn--close--transparent alert--btn__dismiss btn--circular"
-				>
-					{bxIcons.cross}
-				</button>
-			</div>
+		// <div className="alerts flex-col-align-center fw">
```
Comment (saurabhchirde) [inline src/components/UI/Alert/Alert.jsx:29]: remove commented code if not going to use it.

File: src/components/UI/Header/CTA/CTA.jsx [hunk 3]
```diff
@@ -2,16 +2,23 @@ import "./CTA.css";
 import { Link } from "react-router-dom";
 import { useLocation } from "react-router-dom";
 import { ThemeToggle } from "../../ThemeToggle/ThemeToggle";
-import { useAuth } from "../../../../contexts";
+import { useAlert, useAuth } from "../../../../contexts";
```
Comment (saurabhchirde) [inline src/components/UI/Header/CTA/CTA.jsx:5]: Try to give absolute path

File: src/contexts/videoContext.js [hunk 4]
```diff
@@ -162,13 +183,52 @@ export const VideoProvider = ({ children }) => {
...
+				resKey: "history",
+				alert: "Added to history",
+				data: { video: video },
+			});
+			videoDispatch({
+				type: "POST_HISTORY_VIDEOS",
+				payload: output,
+			});
+		} catch (error) {
+			console.log("from addToHistoryVideos", error);
```
Comment (saurabhchirde) [inline src/contexts/videoContext.js:201]: remove console logs

File: src/contexts/videoContext.js [hunk 5]
```diff
@@ -162,13 +183,52 @@ export const VideoProvider = ({ children }) => {
...
+				method: "DELETE",
+				url: removeFromHistoryVideosURL,
+				resKey: "history",
+			});
+			videoDispatch({
+				type: "REMOVE_HISTORY_VIDEOS",
+				payload: output,
+			});
+		} catch (error) {
+			console.log("from removeFromHistoryVideos", error);
```
Comment (saurabhchirde) [inline src/contexts/videoContext.js:219]: this as well

File: src/customHooks/useAxios.js [hunk 6]
```diff
@@ -1,29 +1,46 @@
...
+			}
 			const res = await axios({ url, method, data, headers });
 			if (res.status === 200 || res.status === 201) {
 				response = res.data;
 				output = res.data[resKey];
+				if (!alert) {
+					toggleLoader();
+				}
+				if (alert) {
+					console.log({ alert });
```
Comment (saurabhchirde) [inline src/customHooks/useAxios.js:27]: remove console log

File: src/customHooks/useAxios.js [hunk 7]
```diff
@@ -1,29 +1,46 @@
...
+						...a,
+						visibility: true,
+						text: alert,
+						type: "alert--success",
+					}));
+				}
+			}
+		} catch (err) {
+			error = err.response.data.errors[0];
+			console.log(error);
```
Comment (saurabhchirde) [inline src/customHooks/useAxios.js:38]: remove console log

File: src/pages/Authentication/Login/Login.jsx [hunk 8]
```diff
@@ -6,27 +6,20 @@ import {
 	InputTypeThree,
 	InputTypeTwo,
 } from "../../../components";
-import { useUser, useAuth, useLoader } from "../../../contexts";
+import { useUser, useAuth, useLoader, useAlert } from "../../../contexts";
 import { useDocumentTitle } from "../../../customHooks";
 import "./Login.css";
 import { signIn } from "../../../services";
 import { bxIcons } from "../../../data/icons";
```
Comment (saurabhchirde) [inline src/pages/Authentication/Login/Login.jsx:13]: Absolute path

**Review by saurabhchirde — APPROVED** [review]
Gave some suggestions, others than that couldn't find any issue. Good work on the UI part.
Looks good to merge.

**Review by yashodaAgarwal — APPROVED** [review]
Good job with the UI, your code is structured and neat. Have sme suggestion other than that your code looks good to go you can go ahead and merge!

File: src/contexts/videoContext.js (hunk 4 above)
Comment (yashodaAgarwal) [inline src/contexts/videoContext.js:201]: Please try using alert or toast display instead of console log

**Review by yashodaAgarwal — APPROVED** [review]

File: src/components/UI/Header/CTA/CTA.jsx (hunk 3 above)
Comment (PuneetP16) [reply src/components/UI/Header/CTA/CTA.jsx:5]: Nice suggestion will do that

File: src/components/UI/Alert/Alert.jsx (hunk 2 above)
Comment (PuneetP16) [reply src/components/UI/Alert/Alert.jsx:29]: Ok

File: src/contexts/videoContext.js (hunk 4 above)
Comment (PuneetP16) [reply src/contexts/videoContext.js:201]: ok

File: src/contexts/videoContext.js (hunk 5 above)
Comment (PuneetP16) [reply src/contexts/videoContext.js:219]: Kept that to tap error in case of wrong data input

File: src/customHooks/useAxios.js (hunk 6 above)
Comment (PuneetP16) [reply src/customHooks/useAxios.js:27]: ok

File: src/customHooks/useAxios.js (hunk 7 above)
Comment (PuneetP16) [reply src/customHooks/useAxios.js:38]: ok

File: src/pages/Authentication/Login/Login.jsx (hunk 8 above)
Comment (PuneetP16) [reply src/pages/Authentication/Login/Login.jsx:13]: will do that

File: src/contexts/videoContext.js (hunk 4 above)
Comment (PuneetP16) [reply src/contexts/videoContext.js:201]: I don't want to show this error to the user, its for only development purpose. will remove it in final draft

**Comment by PuneetP16** [comment]
Thanks for the Review!

**Pull request merged**